packages/drivers/odsp-driver/src/epochTracker.ts

```diff
@@ -187,17 +183,37 @@ export class EpochTracker {
         fromCache: boolean = false,
     ) {
         if (error.errorType === OdspErrorType.epochVersionMismatch) {
-            const err = {
-                ...error,
-                fromCache,
-                clientEpoch: this.fluidEpoch,
-                serverEpoch: epochFromResponse ?? undefined,
-                fetchType,
-            };
-            this.logger.sendErrorEvent({ eventName: "EpochVersionMismatch" }, err);
-            assert(!!this.fileEntry, "File Entry should be set to clear the cached entries!!");
-            // If the epoch mismatches, then clear all entries for such file entry from cache.
-            await this.persistedCache.removeEntries(this.fileEntry);
+            try {
+                // This will only throw if it is an epoch error.
+                this.checkForEpochErrorCore(epochFromResponse, error.errorMessage);
```

Comment:
It's a bit more refactoring (which is likely better to do in separate PR), but I think code would be easier to read if
1. checkForEpochErrorCore() would not throw but return an error object
2. Which suggest that most of the logic in throwOdspNetworkError should be moved into createOdspNetworkError and making throwOdspNetworkError  very shellow, and thus createOdspNetworkError() being used directly from checkForEpochErrorCore

What do you think?

Reply:
Yes, I agree createOdspNetworkError  should do most of the work for creating the error and throwOdspNetworkError maybe just throw that error returned by create function. Currently both of them kind of add properties.

Reply:
I will take that up in other PR.